### DISCUSSION TITLE
How to add new IKEA Store?

### DISCUSSION THREAD

**Discussion by observatueur** [Ephigenia/ikea-availability-checker#64] [Q&A] [answered]
Hi,

how to add new IKEA Store?

Regards,

**Comment by Ephigenia** [comment, marked as answer]
new stores can be added following the format in: https://github.com/Ephigenia/ikea-availability-checker/blob/master/source/data/stores.json

**Comment by observatueur** [comment]
Thanks a lot! Where can I find IKEA official stores list?

**Reply by Ephigenia** [reply]
You can list all stores and additional information like geo-locations using the securema2 API like this. Please note that this might not work for some countries. I don't know when it works and when it fails.

```
curl --location --request GET 'https://securema2.ikea.com/stores/v1/us/en/' \
  --header 'Contract: 37249' \
  --header 'Consumer: IRAA'
```

Another way is to reverse engineer the product detail page to find the `store_id`:
<img width="2217" alt="Bildschirmfoto 2021-10-15 um 10 04 45" src="https://user-images.githubusercontent.com/73654/137454311-5c8fc0fb-d04d-4c9f-ae3a-3a4d1260ad76.png">

This information can be added to the `stores.json` file then.
